opta/commands/deploy.py

```diff
@@ -156,3 +160,17 @@ def deploy(
         image_digest=image_digest,
         detailed_plan=detailed_plan,
     )
+
+
+def __check_layer_and_image(layer: "Layer", image: str) -> str:
```

Comment:
I would suggest to keep this super simple by removing the layer, something like
```
image_name, is_custom = __check_image(image_from_config: str, image_from_option: str)
# these names make the code easier to read
```
we would only compute `is_custom` in this function and use the boolean var then